File: leetcode/0014.go
package leetcode

func longestCommonPrefix(strs []string) string {
    if len(strs) == 0{
        return ""
    }
    minstr, maxstr := strs[0],strs[0]
    for _,s := range strs{
        if s < minstr{
            minstr = s
        }
        if s > maxstr{
            maxstr = s
        }
    }
    i := 0
    for ; i < len(minstr)&&minstr[i]==maxstr[i];i++{
    }

    return minstr[:i]
}

File: leetcode/0014.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string> &strs) {
        if (strs.size() == 0) return "";
        string tmin = strs[0], tmax = strs[0];
        for (auto i : strs) {
            tmin = min(tmin, i);
            tmax = max(tmax, i);
        }
        int i = 0;
        for (; i < tmin.length() && tmin[i] == tmax[i]; ++i) {
        }
        return tmin.substr(0, i);
    }
};